crates/beerus-rpc/src/lib.rs

```diff
@@ -372,32 +374,31 @@ impl BeerusApiServer for BeerusRpc {
     }
 
     async fn pending_transactions(&self) -> Result<Vec<Transaction>, Error> {
-        let transactions_result = self
-            .beerus
+        self.beerus
             .starknet_lightclient
             .pending_transactions()
             .await
-            .map_err(|_| Error::from(BeerusApiError::FailedToFetchPendingTransactions));
-        Ok(transactions_result.unwrap())
+            .map_err(|_| Error::from(BeerusApiError::FailedToFetchPendingTransactions))
     }
 
     async fn estimate_fee(
         &self,
         block_id: BlockId,
-        broadcasted_transaction: String,
+        broadcasted_transaction: BroadcastedTransaction,
     ) -> Result<FeeEstimate, Error> {
-        let broadcasted_transaction: BroadcastedTransaction =
-            serde_json::from_str(&broadcasted_transaction).map_err(|e| {
-                Error::Call(CallError::InvalidParams(anyhow::anyhow!(e.to_string())))
-            })?;
-
-        let estimate_fee = self
-            .beerus
+        self.beerus
             .starknet_lightclient
             .estimate_fee(broadcasted_transaction, &block_id)
             .await
-            .map_err(|e| Error::Call(CallError::Failed(anyhow::anyhow!(e.to_string()))))?;
-        Ok(estimate_fee)
+            .map_err(|e| {
+                let error_type: String = e.to_string();
+                match error_type.as_str() {
+                    BLOCK_NOT_FOUND => Error::from(BeerusApiError::BlockNotFound),
+                    CONTRACT_ERROR => Error::from(BeerusApiError::ContractError),
+                    CONTRACT_NOT_FOUND => Error::from(BeerusApiError::ContractNotFound),
+                    _ => Error::from(BeerusApiError::ContractError),
```

Comment:
Let me know what you think of this, I could replicate it in methods that throw more than one error, but the way to do it is open to change.

Reply:
good for now